### PR TITLE
Update capability names as they are named in IRCv3.2.

### PR DESCRIPTION
**Pull request by DarthGandalf**
Old names will be available for a while, then removed.
self-message is superseded by echo-message and will be removed in future
too.

Also remove CAP CLEAR, because it's useless, and got deprecated in
IRCv3.1.

### CODE REVIEW

File: src/Client.cpp [hunk 1]
```diff
@@ -874,15 +874,24 @@ void CClient::HandleCap(const CString& sLine)
 	//TODO support ~ and = modifiers
 	CString sSubCmd = sLine.Token(1);
 
+	std::map<CString, std::function<void(bool bVal)>> mCoreCaps = {
```
Comment (jpnurmi) [inline src/Client.cpp]: Is std::function not overkill for setting a boolean? :)

File: src/Client.cpp [hunk 2]
```diff
@@ -948,46 +948,6 @@ void CClient::HandleCap(const CString& sLine)
 	} else if (sSubCmd.Equals("LIST")) {
 		CString sList = CString(" ").Join(m_ssAcceptedCaps.begin(), m_ssAcceptedCaps.end());
 		RespondCap("LIST :" + sList);
-	} else if (sSubCmd.Equals("CLEAR")) {
```
Comment (jpnurmi) [inline src/Client.cpp]: "Also ..." in a commit message is often a good indication that it could/should be a separate commit.

File: src/Client.cpp (hunk 1 above)
Comment (DarthGandalf) [inline src/Client.cpp]: I just wanted to decrease code duplication... Now this list is listed 4 times (including CLEAR)

File: src/Client.cpp (hunk 2 above)
Comment (DarthGandalf) [inline src/Client.cpp]: Done

File: src/Client.cpp (hunk 1 above)
Comment (jpnurmi) [inline src/Client.cpp]: Yep, that's good. I was just thinking that the map value could've been a simple bool ptr/ref, but perhaps there will be caps with more complex handler functions in the future.

**Comment by jpnurmi** [comment]
+1

**Comment by jpnurmi** [comment]
Related/conflicting cap-notify mockup: https://github.com/jpnurmi/znc/commit/48d0c822dd013075b5ce834d166cd123a04d5d44
